packages/expo-image/src/ExpoImage.web.tsx

```diff
@@ -1,8 +1,89 @@
 import React from 'react';
-import { Image } from 'react-native';
 
-export default function ExpoImage({ source, ...props }) {
-  const resolvedSource = source ?? {};
+import { ImageContentPosition, ImageProps, ImageSource } from './Image.types';
+import { resolveContentFit, resolveContentPosition } from './utils';
 
-  return <Image {...props} source={resolvedSource} />;
+const resolveAssetSource = (source: ImageSource | string | number | undefined | null) => {
+  if (source == null) return null;
+
+  if (typeof source === 'string') {
+    return { uri: source };
+  }
+  if (typeof source === 'number') {
+    return { uri: String(source) };
+  }
+
+  return source;
+};
+
+const ensureUnit = (value: string | number) => {
+  const trimmedValue = String(value).trim();
+  if (trimmedValue.endsWith('%')) {
+    return trimmedValue;
+  }
+  return `${trimmedValue}px`;
+};
+
+const getObjectPositionFromContentPosition = (contentPosition?: ImageContentPosition) => {
+  const resolvedPosition =
+    typeof contentPosition === 'string' ? resolveContentPosition(contentPosition) : contentPosition;
+  if (!resolvedPosition) return undefined;
```

Comment:
As we discussed, let's try not to use `undefined` explicitly
```suggestion
  if (!resolvedPosition) {
    return null;
  }
```

Reply:
This returns undefined since it get's passed directly to the state object which expects either a string or undefined, but we can always do `getObjectPositionFromContentPosition(contentPosition) || undefined`